### PR TITLE
refactor(storage): deprecate Bucket.create() and requester_pays

### PR DESCRIPTION
**Pull request by IlyaFaer**
Towards #7762 

Completing steps 1, 2 and 3 from the [design doc](https://docs.google.com/document/d/1A4FIxThZK_enK7OV9ChY54IPrlpbdHwuhvVEu6iTHxM/edit?usp=sharing):

image

### CODE REVIEW

File: storage/tests/unit/test_bucket.py [hunk 1]
```diff
@@ -607,217 +607,6 @@ def api_request(cls, *args, **kwargs):
         expected_cw = [((), expected_called_kwargs)]
         self.assertEqual(_FakeConnection._called_with, expected_cw)
 
-    def test_create_w_user_project(self):
```
Comment (IlyaFaer) [inline storage/tests/unit/test_bucket.py]: Most part of these tests had already been move into `client_test.py`. Some will be moved with this PR

File: storage/tests/unit/test_client.py [hunk 2]
```diff
@@ -660,7 +751,7 @@ def test_create_bucket_w_string_success(self):
         http = _make_requests_session([_make_json_response(data)])
         client._http_internal = http
 
-        bucket = client.create_bucket(bucket_name, requester_pays=True)
```
Comment (IlyaFaer) [inline storage/tests/unit/test_client.py]: Arg is now deprecated

File: storage/google/cloud/storage/bucket.py [hunk 3]
```diff
@@ -718,43 +718,20 @@ def create(
...
-            path="/b",
-            query_params=query_params,
-            data=properties,
-            _target_object=self,
+        client.create_bucket(
+            bucket_or_name=self,
+            project=project,
+            location=location,
+            predefined_acl=predefined_acl,
+            predefined_default_object_acl=predefined_default_object_acl,
```
Comment (IlyaFaer) [inline storage/google/cloud/storage/bucket.py]: Deleting old implementation and mirroring to `Client.create_bucket()`

**Comment by tswast** [comment]
Are there any code samples (here or in https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/storage) that still use these methods? If so, we should wait to deprecate until those are updated.

Also, we should make sure every API request you can make on Bucket can also be done from Client. It'll make for a much clearer developer story when Bucket can be treated as a dumb resource object.

**Comment by IlyaFaer** [comment]
@tswast, yeah, I've checked python-docs-samples and didn't find any mentions of `bucket.create()` and `requester_pays` arg, only `client.create_bucket()`.
> Also, we should make sure every API request you can make on Bucket can also be done from Client. It'll make for a much clearer developer story when Bucket can be treated as a dumb resource object.

Well, I'm following the design doc for now, pushing piece by piece, as it's a lot of changes. If we need to mirror calls from `Client()` to `Bucket()`, maybe we should try to play around with `getattr()` and just make one redirecting function?

**Comment by tswast** [comment]
> maybe we should try to play around with getattr() and just make one redirecting function?

In my opinion, piece-by-piece as you're currently doing will make it easiest for others to contribute / onboard new teammembers.

**Comment by IlyaFaer** [comment]
Friendly ping. This PR is waiting for review

**Comment by plamut** [comment]
Please reopen in the [new repo](https://github.com/googleapis/python-storage/pull/43), thanks!

**Comment by IlyaFaer** [comment]
Moved to https://github.com/googleapis/python-storage/pull/49

**Pull request closed without merging**